openpgp/packet/public_key.go

```diff
@@ -991,6 +992,33 @@ func (pk *PublicKey) VerifyUserIdHashTag(id string, sig *Signature) (err error)
 	return VerifyHashTag(preparedHash, sig)
 }
 
+// userAttributeSignatureHash returns a Hash of the message that needs to be signed
+// to assert that pk is a valid key for id.
+func userAttributeSignatureHash(uat *UserAttribute, pk *PublicKey, hashFunc crypto.Hash) (h hash.Hash, err error) {
```

Comment:
Please add this function a bit below, above `VerifyUserAttributeSignature`. Also, the comment needs to be updated:
```suggestion
// userAttributeSignatureHash returns a Hash of the message that needs to be signed
// to assert that pk is a valid key for uat.
func userAttributeSignatureHash(uat *UserAttribute, pk *PublicKey, hashFunc crypto.Hash) (h hash.Hash, err error) {
```